### ISSUE TITLE
.has-error, .has-success No Longer Valid for Bootstrap 4

### ISSUE THREAD

**Issue by Joyrex** [berkan52/select2-bootstrap4-theme#10]
changing all instances of `.has-error` to `.is-invalid` and `.has-success` to `.is-valid` allows validation classes to properly render on Select2 elements.

**Comment by VictorioBerra** [comment]
[Replaced `.has-error`, `.has-warning`, and `.has-success` classes with HTML5 form validation via CSS’s `:invalid` and `:valid` pseudo-classes.](https://getbootstrap.com/docs/4.1/migration/)

We should definitely fix this. PRs welcome.

**Comment by VictorioBerra** [comment]
For me, It looks like jQuery DataTables is adding has-error still. Not sure if they have fixed that.

**Comment by Joyrex** [comment]
Yeah, DataTables needs to update their Bootstrap 4 styling to reflect 4.1.1 - it wouldn't hurt to leave the legacy classes in there; you'd just have extra CSS.

**Comment by VictorioBerra** [comment]
DataTables fixed this recently https://editor.datatables.net/download/1.7.4
